src/LibraryManager/SemanticVersion/SemanticVersion.cs

```diff
@@ -186,24 +193,92 @@ public int CompareTo(SemanticVersion other)
             return StringComparer.OrdinalIgnoreCase.Compare(OriginalText, other.OriginalText);
         }
 
+        /// <summary>
+        /// Returns whether the semantic verisons are equal.  This includes comparing the build metadata, and does not provide semantic equivalence.
+        /// </summary>
         public bool Equals(SemanticVersion other)
         {
             return other != null && string.Equals(OriginalText, other.OriginalText, StringComparison.OrdinalIgnoreCase);
         }
 
+        /// <summary>
+        /// Returns whether the other object is equal to this SemanticVersion.
+        /// </summary>
         public override bool Equals(object obj)
         {
             return Equals(obj as SemanticVersion);
         }
 
+        /// <summary>
+        /// Returns a hash code to uniquely identify this version.
+        /// </summary>
+        /// <remarks>
+        /// This is aware of build metadata, and should not be relied on as a semantic equality comparison.
+        /// </remarks>
         public override int GetHashCode()
         {
             return _hashCode;
         }
 
+        /// <summary>
+        /// Get a string representation of this SemanticVersion
+        /// </summary>
         public override string ToString()
         {
             return OriginalText;
         }
+
+        /// <summary>
+        /// Equality operator
+        /// </summary>
+        public static bool operator ==(SemanticVersion left, SemanticVersion right)
+        {
+            if (ReferenceEquals(left, null))
+            {
+                return ReferenceEquals(right, null);
+            }
+
+            return left.Equals(right);
+        }
+
+        /// <summary>
+        /// Inequality operator
+        /// </summary>
+        public static bool operator !=(SemanticVersion left, SemanticVersion right)
+        {
+            return !(left == right);
+        }
+
+        /// <summary>
+        /// Less-than operator
+        /// </summary>
+        public static bool operator <(SemanticVersion left, SemanticVersion right)
+        {
+            return ReferenceEquals(left, null) ? !ReferenceEquals(right, null) : left.CompareTo(right) < 0;
```

Comment:
Out of curiosity, why `ReferenceEquals(left/right, null)` and not `left/right is null`? Same on the other lt/gt operators.

Reply:
That was generated from the suggested code fix, but I can change it to use `is null` instead.